### PR TITLE
Porting app folder to TypeScript by transpiling to another folder

### PR DESCRIPTION
**Pull request by LabhanshAgrawal**
Reference - #3836 
This PR tries to port app folder to TS by copying assets & transpiling to compiled_app folder and using that for building and running Hyper.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # build output
 dist
 app/renderer
+compiled_app
```
Comment (ppot) [inline .gitignore]: target

**Comment by Stanzilla** [comment]
@LabhanshAgrawal hey there, what's the status on this? what's missing?

**Comment by LabhanshAgrawal** [comment]
I'm facing problems with json modules while bundling.

**Comment by LabhanshAgrawal** [comment]
Hey @Stanzilla I'm not able to get the bundling to work. I tried some approaches but they either give errors during webpack or launch.
This is working in it's current form also. If we need bundling it will take more time probably. Or if it is ok, we can merge this and add bundling later.

**Comment by Stanzilla** [comment]
Sounds fine to me, I'm having the weirdest problem with the Electron branch as well atm, so that will probably take a while to fix. 

What do the others think about this? Would be nice to get some testing on Linux/Mac here since I don't have those. @ivanwonder @GitSquared

**Comment by GitSquared** [comment]
@Stanzilla I haven't looked into what this PR changes but cloning it and running from source works flawlessly, on Linux x64.

**Comment by Stanzilla** [comment]
@LabhanshAgrawal slight problem that I only noticed after merging, there is no `cp -r` on Windows so we will have to use something else here

**Comment by LabhanshAgrawal** [comment]
@Stanzilla looking into it

**Comment by Stanzilla** [comment]
@LabhanshAgrawal I tried cpy-cli here https://github.com/zeit/hyper/pull/3785/commits/fc47596242bd17aa24b371f894dbe034e8f887ee

**Comment by LabhanshAgrawal** [comment]
#3852

**Pull request merged**